src/controllers/cart.controller.ts

```diff
@@ -0,0 +1,143 @@
+import * as Koa from 'koa'
+import * as HttpStatus from 'http-status-codes'
+import { getRepository, Repository } from 'typeorm'
+import CartPosition from '../models/cart.entity'
+import User from '../models/user.entity'
+import Dish from '../models/menu.entity'
+import Order from '../models/order.entity'
+
+export default class CartController {
+    static async getCart(ctx: Koa.Context){
+        const cartRepo: Repository<CartPosition> = getRepository(CartPosition)
+        const userRepo: Repository<User> = getRepository(User)
+        const orderRepo: Repository<Order> = getRepository(Order)
+
+        const user: User = await userRepo.findOne(ctx.params.user_id)
+        let order: Order = await orderRepo.findOne({
+            where: {
+                user: user,
+                state: 'Initial',
+            }
+        })
+        if (!order) ctx.throw(HttpStatus.NOT_FOUND, 'User has no order')
+        let cart: CartPosition[] = await cartRepo.find({
+            where: {
+                order: order
+            },
+            relations: ['dish'],
+        })
+        ctx.body = {
+            cart
+        }
+    }
+
+    static async addDishToCart(ctx: Koa.Context){
+        const cartRepo: Repository<CartPosition> = getRepository(CartPosition)
+        const userRepo: Repository<User> = getRepository(User)
+        const orderRepo: Repository<Order> = getRepository(Order)
+        const menuRepo: Repository<Dish> = getRepository(Dish)
+
+        const user: User = await userRepo.findOne(ctx.params.user_id)
+        let order: Order = await orderRepo.findOne({
+            where: {
+                user: user,
+                state: 'Initial',
+            }
+        })
+        if (!order) {
+            const newOrder: Order = orderRepo.create({
+                price: 0,
+                state: 'Initial',
+                user: user,
+                cartPositions: []
+            })
+            await orderRepo.save(newOrder)
+            order = newOrder
+        }
+        const dish: Dish = await menuRepo.findOne(ctx.request.body.dish_id)
+        if(!dish) ctx.throw(HttpStatus.NOT_FOUND, 'Dish is not found')
+        const cartPosition: CartPosition = cartRepo.create({
+            dish: dish,
+            ingredients: ctx.request.body.ingredienst || dish.ingredients,
+            quantity: ctx.request.body.quantity || 1,
+            order: order,
+        })
+        await cartRepo.save(cartPosition)
```

Comment:
The issue of this whole endpoint is that it does too many requests that in addition are not in transaction, so it could cause a lot of data inconsistency. For example, if user adds 2 dishes in parallel, it could create 2 carts.
Most likely the amount of requests could be decreased with the approach like I described above and with things like upsert/partial update. Unfortunately I'm not fully familiar with typeorm to be able to help with the exact approach here. But you can start with at least wrapping everything in transaction to prevent data corruption.